### PR TITLE
Dev nicolson

### PR DESCRIPTION
**Pull request by trinisofttechnologies**
Checking my changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,9 @@
   "bugs": {
     "url": "https://github.com/Countly/countly-sdk-react-native-bridge/issues"
   },
+  "scripts": {
+    "postinstall": "node ./hooks/createNotificationService.js"
```
Comment (ArtursKadikis) [inline package.json:13]: what does this do?

File: example/Example.js [hunk 2]
```diff
@@ -291,14 +292,29 @@ class Example extends Component {
       var latitude = "29.634933";
       var longitude = "-95.220255";
       var ipAddress = "103.238.105.167";
+      Countly.setLocation(null, city, latitude + "," + longitude, ipAddress);
+      Countly.setLocation(null, null, latitude + "," + longitude, ipAddress);
```
Comment (ArtursKadikis) [inline example/Example.js:296]: remove the wrongly used samples

File: ios/src/CountlyReactNative.m [hunk 3]
```diff
@@ -358,31 +375,41 @@ + (void)onNotification:(NSDictionary *)notification
 
 RCT_EXPORT_METHOD(setLocation:(NSArray*)arguments)
 {
-  NSString* countryCode = [arguments objectAtIndex:0];
-  NSString* city = [arguments objectAtIndex:1];
-  NSString* location = [arguments objectAtIndex:2];
-  NSString* IP = [arguments objectAtIndex:3];
+  NSString* city = [arguments objectAtIndex:0];
+  NSString* country = [arguments objectAtIndex:1];
```
Comment (ArtursKadikis) [inline ios/src/CountlyReactNative.m:379]: The order of values read is wrong, country code is first and city is second

File: example/install.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+rm -rf ./AwesomeProject
+react-native init AwesomeProject
```
Comment (ArtursKadikis) [inline example/install.sh:2]: What's the point of this script? add a comment

File: example/install.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+rm -rf ./AwesomeProject
```
Comment (ArtursKadikis) [inline example/install.sh:1]: The script is missing a Shebang at the top

File: example/AppDelegate+notification.m [hunk 6]
```diff
@@ -0,0 +1,184 @@
+#import <objc/runtime.h>
```
Comment (ArtursKadikis) [inline example/AppDelegate+notification.m:1]: What is this file doing?

File: hooks/createNotificationService.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+var config = {};
+
+function createNotificationService(config){
+
+};
```
Comment (ArtursKadikis) [inline hooks/createNotificationService.js:5]: Unneccesarry semi colon

File: example/Example.js (hunk 2 above)
Comment (trinisofttechnologies) [reply example/Example.js:296]: These are not samples, these are test cases to see if the sdk is not crashing.

File: example/install.sh (hunk 4 above)
Comment (trinisofttechnologies) [reply example/install.sh:2]: You asked for a script to auto install the example app, this is it.

File: hooks/createNotificationService.js (hunk 7 above)
Comment (trinisofttechnologies) [reply hooks/createNotificationService.js:5]: Ok, I will remove it.

But it is a good javascript practice to have semi colon in method.

File: ios/src/CountlyReactNative.m (hunk 3 above)
Comment (trinisofttechnologies) [reply ios/src/CountlyReactNative.m:379]: Oh, sorry for that.

File: package.json (hunk 1 above)
Comment (trinisofttechnologies) [reply package.json:13]: This isn't doing anything at the moment.

I will remove it.

File: example/AppDelegate+notification.m (hunk 6 above)
Comment (trinisofttechnologies) [reply example/AppDelegate+notification.m:1]: This is not in the sdk at the moment.

This is the route to fix the push notification issue, that the current sdk is missing.

Like Erdinc issue.

File: example/install.sh (hunk 5 above)
Comment (ArtursKadikis) [reply example/install.sh:1]: In case you do no know:
https://en.wikipedia.org/wiki/Shebang_(Unix)

File: hooks/createNotificationService.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+var config = {};
```
Comment (ArtursKadikis) [inline hooks/createNotificationService.js:1]: This empty file should be removed

File: Countly.js [hunk 9]
```diff
@@ -472,10 +445,10 @@ Countly.updateRemoteConfigExceptKeys = function(keyNames, callback){
 
 Countly.getRemoteConfigValueForKey = function(keyName, callback){
     CountlyReactNative.getRemoteConfigValueForKey([keyName.toString() || ""], (value) => {
```
Comment (mksglu) [inline Countly.js:447]: if `keyName` is not required, maybe we can define it as the default parameter and we have to check the type of some values such as String or Number in a separate function.

File: Countly.js [hunk 10]
```diff
@@ -181,25 +169,10 @@ Countly.demo = function(){
...
-        newStringLocation = newStringLocation+""+locationArray[0]
-    }
-    if(locationArray[1].indexOf(".") == -1){
-        newStringLocation = newStringLocation+","+ parseFloat(locationArray[1]).toFixed(2);
-    }else{
-        newStringLocation = newStringLocation+","+locationArray[1]
-    }
-    args.push(newStringLocation);
-    args.push(ipAddress || "0.0.0.0");
+    args.push(countryCode || "null");
```
Comment (mksglu) [inline Countly.js:172]: what exactly is `null` and why a `String` of type is defined?

File: example/Example.js (hunk 2 above)
Comment (mksglu) [reply example/Example.js:296]: Maybe you can use Template String for more readable code.

File: Countly.js (hunk 10 above)
Comment (ArtursKadikis) [reply Countly.js:172]: As far as I understand, the "null" thing is the workaround of how to pass a null value over to the native side. Maybe a limitation of the argument array.

File: Countly.js (hunk 9 above)
Comment (ArtursKadikis) [reply Countly.js:447]: In this case "keyName" is a mandatory parameter.

Comment (mksglu) [reply Countly.js:447]: OK. I see the function is `getRemoteConfigValueForKey` which passed `keyName` and that's a required parameter. In that case, we shouldn't use the logical `OR` operator in the params. I think in the JavaScript side, all mandatory parameters should be checked in a separate function, and empty parameters should be checked before being passed to the Native direction.

File: example/Example.js (hunk 2 above)
Comment (trinisofttechnologies) [reply example/Example.js:296]: Ok, I understand.

Comment (trinisofttechnologies) [reply example/Example.js:296]: Done

File: example/AppDelegate+notification.m (hunk 6 above)
Comment (trinisofttechnologies) [reply example/AppDelegate+notification.m:1]: This file is not a part of sdk, 

But a sample file for reference.

It is located in example folder.

File: Countly.js (hunk 9 above)
Comment (ArtursKadikis) [reply Countly.js:447]: Sounds like a reasonable improvement. But we'll add it in a future refactor

**Pull request merged**